pkg/agent/handler/managedapplication.go

```diff
@@ -31,8 +31,8 @@ func NewManagedApplicationHandler(prov managedAppProvision, cache agentcache.Man
 }
 
 // Handle processes grpc events triggered for ManagedApplications
-func (h *managedApplication) Handle(action proto.Event_Type, _ *proto.EventMeta, resource *v1.ResourceInstance) error {
-	if resource.Kind != mv1.ManagedApplicationGVK().Kind || h.prov == nil || action == proto.Event_SUBRESOURCEUPDATED {
+func (h *managedApplication) Handle(action proto.Event_Type, meta *proto.EventMeta, resource *v1.ResourceInstance) error {
+	if resource.Kind != mv1.ManagedApplicationGVK().Kind || h.prov == nil || isNotStatusSubResourceUpdate(action, meta) {
```

Comment:
How come you want to process some sub resource events? You should be getting an event for the AccessRequest when the sub resource is added to the main resource which makes processing sub resources redundant.

Reply:
What about when a subresource is just updated, not added to a resource?

Reply:
We will certainly get update event since the resource will be created by SaaS controller but we need to handle the sub resource update event as well since the resources can be created using API/CLI (without subresource) and then have an update call to update the subresource. 

Also, for the case where the agent fails the resource provisioning it is going to set the status as Error, but if someone wants to reprocess, they could update the subresource(there should be some sort of state transition validation though I believe that should done)

Reply:
I still don't think it is necessary to handle a subresource event because we fetch the full resource each time we start processing events. So if there was an update to the status subresource we will still fetch the full AccessRequest which will then have all the sub resources, and then we will process it as a normal AccessRequest update.

Reply:
Correct, we fetch the the full resource every time for all events (except for delete event), however SUBRESOURCEUPDATE is raised separately. If the resource itself is not modified but only the subresource is updated using PUT call to /resource/subresource, it will only raise SUBRESOURCEUPDATE event, there will be no UPADTE event for the resource itself.
So we need to handle the subresource event for those cases which is going to be the usual case when these subresoruces will be updated.